packages/core/src/hooks/useUpdateNodeInternals.ts

```diff
@@ -6,12 +6,18 @@ import type { UpdateNodeInternals } from '../types';
 function useUpdateNodeInternals(): UpdateNodeInternals {
   const store = useStoreApi();
 
-  return useCallback<UpdateNodeInternals>((id: string) => {
+  return useCallback<UpdateNodeInternals>((id: string | string[]) => {
     const { domNode, updateNodeDimensions } = store.getState();
     const nodeElement = domNode?.querySelector(`.react-flow__node[data-id="${id}"]`) as HTMLDivElement;
 
     if (nodeElement) {
-      requestAnimationFrame(() => updateNodeDimensions([{ id, nodeElement, forceUpdate: true }]));
+      const updateIds = Array.isArray(id) ? id : [id];
+
+      requestAnimationFrame(() => {
+        updateIds.forEach((id) => {
+          updateNodeDimensions([{ id, nodeElement, forceUpdate: true }]);
```

Comment:
the nodeElement would always be the same node. You also need to call `domNode?.querySelector(`.react-flow__node[data-id="${id}"]`)` multiple times. It's also possible to pass an array with ids and nodeElements to `updateNodeDimensions`.

Reply:
Ah yeah, that makes sense - I'll change that
Thanks for keeping an eye out, I didn't think too much doing this PR and it shows 🤣

Reply:
querySelector should go in the `forEach` so that `nodeElement` matches the actual node.

Reply:
Done 👍